src/celeritas/CMakeLists.txt

```diff
@@ -130,8 +130,6 @@ if(CELERITAS_USE_Geant4)
       ext/detail/GeantGeoConverter.cc
     )
     list(APPEND _cg4_libs VecGeom::vecgeom)
-  else()
```

Comment:
On line 128,(might not be visible on github in the default view) there is:
```
if(CELERITAS_USE_VecGeom)
```
should we consistently use:
```
if(CELERITAS_GEO STREQUAL "VecGeom")
```
instead or is there a different semantic here?

Reply:
Yeah, it's a different semantic: this changeset will allow us to build VecGeom but make it not the "core" geometry. In practice I don't think we'll want to do this, but it's sort of the first step toward potentially having core components templated on the geometry type.

Reply:
In that case, I think it might be clearer to rename the old one `CELERITAS_HAS_VecGeom`

Reply:
USE, HAS, HAVE ... we're still "using" it (compiling it) but not necessarily using it for everything. You could say the same thing about CUDA or MPI.

Reply:
I am trying to make a distinction between:
```
CELERITAS_GEO STREQUAL "VecGeom"   -> Used as the main geometry backend
and
CELERITAS_HAS_VecGeom -> Is available/compiled-in (and might be used auxiliary code).
```
(In my humble opinion) "use" implied that it is the utilized from the main part of the project (i.e. here, as a geometry backend for the main Celeritas library".  (And also if the code becomes completely templated on the geometry backend, then which is ever is being "used" is now per se delegated to the user code/project/library)

Reply:
>  You could say the same thing about CUDA or MPI.

The different is that (at least for now :) ) we don't have 2 closely related `CMake` variable that can easily be confused one for the other (esp. since the `use` one used to have/include the semantic of the new `_GEO` variable).

Reply:
I don't want to rename all the `CELERITAS_USE_([A-Z]+)` to `CELERITAS_HAVE_\1`. Do you think it's worth renaming `CELERITAS_(RNG|GEO)` to `CELERITAS_CORE_\1`? We do compile CUDA/HIP RNGs if available even though we might be using the in-house XORWOW.

Reply:
> I don't want to rename all

I agree as all the other are unambiguous.  I was recommending to **only** change the  `CELERITAS_USE_*` to `CELERITAS_HAS_\1`.

But I guess, it would also make sense for the `RNG` ones.

Reply:
It doesn't make any sense to have `CELERITAS_HAS_VecGeom` and `CELERITAS_USE_CUDA` doing the same thing: showing that an optional library is being compiled in.

Reply:
Both. And for now it's build-time, but if we add templates to everything that uses the CoreTrackData, CoreTrackView, or GeoTrackView, it could be a runtime. Our MC code Shift does this but I deliberately said we should avoid that for now for the sake of simplicity.

Reply:
> It doesn't make any sense to have CELERITAS_HAS_VecGeom and CELERITAS_USE_CUDA doing analogous thing: 

That's a good point ... so indeed it would be all or nothing.

> Do you think it's worth renaming `CELERITAS_(RNG|GEO)` to `CELERITAS_CORE_\1`?

maybe ... I guess (eventually) it would be also semantically close to `CELERITAS_DEFAULT_\1`  ....

Reply:
`CELERITAS_DEFAULT_` would I think become `CELERITAS_CORE_DEFAULT_`...

Reply:
Sounds good to me :)